linear.cpp

```diff
@@ -44,6 +47,60 @@ static void info(const char *fmt,...)
 #else
 static void info(const char *fmt,...) {}
 #endif
+
+// New function to ensure the same behaviour for random number generation on windows and linux
+int myrand() {
+#ifdef _WIN32
+	// In MS Visual Studio (2012) RAND_MAX = 0x7FFF (15bit) = 32767
+	// In Linux GCC (4.6) 32bits RAND_MAX = 0x7FFFFFFF (31bit) = 2147483647 
+	// In Linux GCC (4.6) 64bits RAND_MAX = 0x7FFFFFFFFFFFFFFF (63 bits) = 9223372036854775807
+	// so in MS Visual Studio we need to call rand() several times to always ensure the same random number range than in Linux GCC
+	if (std::numeric_limits<int>::max() == 0x7FFFFFFF) {
+		// make a 31bit random number by using several 15bit rand()
+		return ((rand() << 16) + (rand() << 1) + (rand() >> 14));
+	}
+	else if (std::numeric_limits<int>::max() == 0x7FFFFFFFFFFFFFFF) {
```

Comment:
I don't think there are any Windows platforms with a 64-bit `int`.

Reply:
not yet indeed but at least we're prepared :)